packages/driver/src/cy/commands/navigation.js

```diff
@@ -352,7 +352,7 @@ module.exports = (Commands, Cypress, cy, state, config) => {
 
   Cypress.on('test:before:run:async', () => {
     // reset any state on the backend
-    Cypress.backend('reset:server:state')
+    return Cypress.backend('reset:server:state')
```

Comment:
Why? Other `.on` methods don't do this.

Reply:
I do not know exactly but @brian-mann did so in this commit 
https://github.com/cypress-io/cypress/commit/3ff21b91bcd404f67d872191c5233f4d7273505c#diff-2010cae8f580677e575ffcb8c87ccd2091e85d4d40a3cff5b38d28f612dcf92eR356

@brian-mann can you shed a light?

Reply:
This is binding to `test:before:run:async` - which awaits promises before resolving. The others don't do this because they're binding to a sync event, which does not need to be awaited.

Other `test:before:run:async` event handlers are properly returning.

This is necessary because we *must* wait for the server to ack on resetting its state, otherwise the next test could start without this being done yet.